### DISCUSSION TITLE
Contact points between rigid and soft bodies in PyBullet

### DISCUSSION THREAD

**Discussion by AlexanderFabisch** [bulletphysics/bullet3#3367] [General]
I try to implement a simple prototype to grasp a soft body with a robotic hand. The results look quite promising so far. Great work by the way!

Now I want to determine the contact forces between the hand and the soft body. I noticed that `pybullet.getContactPoints(...)` only gives me contact points between rigid bodies. Is it "easily" possible to extent it to include soft bodies?

I looked already a bit into the implementation of the Python function [here](https://github.com/bulletphysics/bullet3/blob/93be7e644024e92df13b454a4a0b0fcd02b21b10/examples/pybullet/pybullet.c#L9695) and its corresponding implementation in the shared memory interface [here](https://github.com/bulletphysics/bullet3/blob/93be7e644024e92df13b454a4a0b0fcd02b21b10/examples/SharedMemory/PhysicsServerCommandProcessor.cpp#L7939). I could invest some time to prepare a pull request, but I might need some hints to start.

**Comment by erwincoumans** [comment]
It is best to create a separate function (getContactPoints2) to avoid breaking existing code. Not sure if the softbody contacts and the forces are stored persistently. Have you looked into it?

**Reply by AlexanderFabisch** [reply]
My colleague @oscar-lima is currently working on it. We will hopefully be able to open a pull request soon.

**Reply by johbrust** [reply]
Hi, I took over from @oscar-lima and I opened a [pull-request](https://github.com/bulletphysics/bullet3/pull/3979) a few minutes ago. It is based on the contributions by pull-requests #3937 and #3949. I added the information on contact forces based on the impulses acting on the tetrahedral nodes.

**Reply by erwincoumans** [reply]
Nice, I'll check it out soon.
